src/derive.md

```diff
@@ -0,0 +1,43 @@
+# Derive
+
+<!-- TODO: Maybe link to https://doc.rust-lang.org/book/appendix-03-derivable-traits.html ?
+Should the reference say more about which traits are derivable?
+-->
+
+The *`derive` attribute* allows certain traits to be automatically implemented
+for data structures. It uses the [_MetaListPaths_] syntax to specify a list of
+traits to implement.
+
+For example, the following will create an `impl` for the
+`PartialEq` and `Clone` traits for `Foo`, and the type parameter `T` will be
+given the `PartialEq` or `Clone` constraints for the appropriate `impl`:
+
+```rust
+#[derive(PartialEq, Clone)]
+struct Foo<T> {
+    a: i32,
+    b: T,
+}
+```
+
+The generated `impl` for `PartialEq` is equivalent to
+
+```rust
+# struct Foo<T> { a: i32, b: T }
+impl<T: PartialEq> PartialEq for Foo<T> {
+    fn eq(&self, other: &Foo<T>) -> bool {
+        self.a == other.a && self.b == other.b
+    }
+
+    fn ne(&self, other: &Foo<T>) -> bool {
+        self.a != other.a || self.b != other.b
+    }
+}
+```
+
+You can implement `derive` for your own traits through [procedural macros].
+
+<!-- TODO: automatically_derived -->
```

Comment:
These are documented [here](https://doc.rust-lang.org/nightly/reference/special-types-and-traits.html#auto-traits), and not part of the derive macro system.

Reply:
This refers to the `#[automatically_derived]` attribute which is added to derived impls.  It may have been accidentally leaked to stable, but now that it is in use by several crates (see https://github.com/rust-lang-nursery/reference/issues/532#issuecomment-471086171) it may need to be documented.